Update ADL to latest Swagger Spec and TraceLogging added

begoldsm opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
- Update to latest Swagger commit (PR: https://github.com/Azure/azure-rest-api-specs/pull/844)
    - Adds Pricing tier support for ADLA and ADLS
    - Fixes bug with missing enum values for severity for job diagnostics and error details
    - Fixes bug with UpdateCredential incorrectly claiming it returns something, when nothing has ever been returned.
    - Removes IResource classification from our sub resources since they do not strictly adhere to the Azure definition of a sub resource.
- Add tests for Pricing Tier 
- Add logging for ADLS Uploader that can be hooked up into the existing TraceLogging framework.
- Add a change log for ADLA and ADLS to be used with initial stable release
- Update package versions to our **first stable release** based on internal sign-off
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/dev/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

Commit:
Updates to include logging

Commit:
Update to use ServiceClientTracing instead.

Commit:
Add change log and pointers to change log for ADL

This is part of updating the versions to our official stable versions as
well.

Update to latest swagger spec commit for ADL

Update ADL to latest Swagger commit

This change includes:
1. Tests for pricing tier changes (from last swagger commit)
2. Addition of three enum values for severity that already exist in the
service
3. Fix a bug with UpdateSecret incorrectly claiming to return something
when it never has.

Missed assembly version update

Thanks checklist!

Commit:
Update to latest swagger for ADL

This change includes:
1. Rename PricingTierType to TierType. This is not a breaking change
because no clients have been published yet with PricingTierType, as it
was caught by PM before going out.
2. Update comments to remove billing connotations

missed file in last commit.

Address CR comments for central log helper

Missed csproj inclusion of TracingHelper.

cormacpayne commented on src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Customizations/DataLakeAnalyticsCustomizationHelper.cs:
```diff
@@ -18,7 +18,7 @@ internal static class DataLakeAnalyticsCustomizationHelper
         /// This constant is used as the default package version to place in the user agent.
```
@begoldsm the license header at the top of this file says the AutoRest version used is 0.14.0.0 rather than 1.0.0.0

cormacpayne commented on src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/Models/SubResource.cs:
```diff
@@ -18,13 +18,12 @@ namespace Microsoft.Azure.Management.DataLake.Analytics.Models
     /// <summary>
     /// The Sub Resource model definition.
     /// </summary>
-    public partial class SubResource : IResource
```
@begoldsm breaking change?

begoldsm replied:
This shouldn't be a breaking change since IResource is empty. We got a bug on this from the ruby dev, indicating that our definition of a SubResource is not the same as the true Azure one, so it was causing breaks in other languages. For C# IResource is an empty class. Additionally, SubResources cannot be created with just an IResource since they have required parameters for us, so there should be no breaks.

cormacpayne commented on src/ResourceManagement/DataLake.Store/Microsoft.Azure.Management.DataLake.Store/Customizations/DataLakeStoreCustomizationHelper.cs:
```diff
@@ -18,7 +18,7 @@ internal static class DataLakeStoreCustomizationHelper
         /// This constant is used as the default package version to place in the user agent.
         /// It should mirror the package version in the project.json file.
         /// </summary>
-        internal const string PackageVersion = "1.0.2-preview";
```
@begoldsm same comment about license header

begoldsm replied:
I have removed the AutoRest versioning in the license header from all the custom classes, since they are not auto-genned.

cormacpayne commented on src/ResourceManagement/DataLake.Store/Microsoft.Azure.Management.DataLake.Store/Generated/Models/SubResource.cs:
```diff
@@ -16,15 +16,13 @@ namespace Microsoft.Azure.Management.DataLake.Store.Models
     /// The Resource model definition for a nested resource.
     /// </summary>
-    public partial class SubResource : IResource
+    public partial class SubResource
```
@begoldsm breaking change?

begoldsm replied:
Same comment, but please let me know if this is incorrect!

cormacpayne reviewed (changes requested):
@begoldsm a couple of minor comments

Commit:
remove auto gen from custom files

Commit:
Add Deprecation for deprecated methods

1. Update the changeLog to reflect the "breaking" changes for ADLA
2. Update version for ADLA to 2.0.0 to properly indicate the breaking
changes.

Correctly updating ChangeLog version for ADLA

begoldsm commented:
@cormacpayne can you take a look and see if I have addressed your comments?

cormacpayne commented on src/ResourceManagement/DataLake.Analytics/changelog.md:
```diff
@@ -0,0 +1,15 @@
+## Microsoft.Azure.Management.DataLake.Analytics release notes
+
+### Changes in 1.0.2
```
@begoldsm should be 2.0.0
